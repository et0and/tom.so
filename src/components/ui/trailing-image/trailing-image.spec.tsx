import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import TrailingImage from "./trailing-image";

describe("<TrailingImage/>", () => {
  it("renders correctly", () => {
    const { container } = render(<TrailingImage />);
    expect(container).toMatchSnapshot();
  });

  it("renders the correct number of AnimatedImage components", () => {
    const { container } = render(<TrailingImage />);
    const images = container.querySelectorAll("img");
    expect(images.length).toBe(20); // As per the component logic
  });

  it("has the correct class names", () => {
    const { container } = render(<TrailingImage />);
    const mainDiv = container.firstChild as HTMLElement;
    expect(mainDiv).toHaveClass(
      "storybook-fix",
      "relative",
      "flex",
      "min-h-80",
      "w-1/2",
    );
  });
});
