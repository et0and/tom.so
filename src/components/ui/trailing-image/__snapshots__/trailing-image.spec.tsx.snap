// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrailingImage/> renders correctly 1`] = `
<div>
  <div
    class="storybook-fix relative flex min-h-80 w-1/2"
  >
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="a-drawing.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="angel.png"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="spiral.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="uptofriends.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="the-museum-without-walls.png"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="models.png"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="microme.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="pkg.gif"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="nothing-market.png"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="nothing-market-poster.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="chimera.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="cover.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="a-drawing.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="angel.png"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="spiral.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="uptofriends.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="the-museum-without-walls.png"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="models.png"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="microme.jpg"
      style="opacity: 0; transform: none;"
    />
    <img
      alt="trail element"
      class="absolute w-44 object-cover"
      src="pkg.gif"
      style="opacity: 0; transform: none;"
    />
  </div>
</div>
`;
